php_phongo.c

```diff
@@ -2459,7 +2459,138 @@ static char* php_phongo_manager_make_client_hash(const char* uri_string, zval* o
 	return hash;
 }
 
-static mongoc_client_t* php_phongo_make_mongo_client(const mongoc_uri_t* uri) /* {{{ */
+static bool php_phongo_extract_handshake_data(zval* zvalue, const char* key, char** value, size_t* value_len)
+{
+	if (Z_TYPE_P(zvalue) != IS_STRING) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected \"%s\" handshake option to be a string, %s given", key, PHONGO_ZVAL_CLASS_OR_TYPE_NAME_P(zvalue));
+		return false;
+	}
+
+	*value     = estrdup(Z_STRVAL_P(zvalue));
+	*value_len = Z_STRLEN_P(zvalue);
+
+	return true;
+}
+
+static char* php_phongo_concat_handshake_data(const char* default_value, const char* custom_value, size_t custom_value_len)
+{
+	char*  ret;
+	size_t ret_len = strlen(default_value) + 1;
+
+	if (custom_value) {
+		ret_len += custom_value_len + 1;
```

Comment:
`ret_len` already covers the null byte, so the `+1` here is for the pipe separator character, correct?

Reply:
Yes, that's correct. I've added a comment to clarify.